### PR TITLE
feat: add proxy in order to fetch trustfolio api

### PR DESCRIPTION
**Pull request by benjyup**

### CODE REVIEW

File: api/index.js [hunk 1]
```diff
@@ -0,0 +1,40 @@
+import fetch from 'node-fetch'
+
+const handler = async (request, response) => {
+  const {endpoint} = request.query
```
Comment (tusbar) [inline api/index.js]: Why pass the endpoint? Are there multiple ones?

Also this is not safe. I’d rather have a `environment` search params that allows to pick which endpoint if there are multiple ones (or something like that).

File: api/index.js [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+    response.status(400).json(out)
+    return
+  }
+
+  try {
+    const result = await fetch(endpoint, {
+      method: 'POST',
+      headers: {
+        'X-Requested-With': 'XMLHttpRequest',
```
Comment (tusbar) [inline api/index.js]: What is this for?

File: api/index.js [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+    }).then((response) => response.json())
+
+    response.status(200).send(result)
+  } catch (error) {
+    const out = {
+      error: 'Something went wrong...',
+      message: error.message,
+    }
+
+    response.status(500).json(out)
```
Comment (tusbar) [inline api/index.js]:
```suggestion
    response.status(500).send({…})
```

File: package.json [hunk 4]
```diff
@@ -25,10 +25,12 @@
     ]
   },
   "dependencies": {
+    "axios": "^0.27.2",
```
Comment (tusbar) [inline package.json]: Why axios?

File: package.json [hunk 5]
```diff
@@ -25,10 +25,12 @@
...
     "datocms-plugin-sdk": "^0.5.2",
     "datocms-react-ui": "^0.5.2",
     "final-form": "^4.20.6",
     "lodash": "^4.17.21",
+    "node-fetch": "2",
```
Comment (tusbar) [inline package.json]: Use `yarn add node-fetch` please

**Review by tusbar — CHANGES_REQUESTED** [review]

File: api/index.js (hunk 1 above)
Comment (benjyup) [reply api/index.js]: No there's no multiple ones indeed

File: utils/index.js [hunk 6]
```diff
@@ -0,0 +1,13 @@
+const filterProperties = (raw, unallowed) => {
```
Comment (tusbar) [inline utils/index.js]: You can remove this too

File: vercel.json [hunk 7]
```diff
@@ -0,0 +1,13 @@
+{
+  "headers": [
+    {
+      "source": "/api/(.*)",
+      "headers": [
+        { "key": "Access-Control-Allow-Credentials", "value": "true" },
+        { "key": "Access-Control-Allow-Origin", "value": "*" },
+        { "key": "Access-Control-Allow-Methods", "value": "GET,OPTIONS,PATCH,DELETE,POST,PUT" },
```
Comment (tusbar) [inline vercel.json]: It seems like we’re only doing POST requests

File: vercel.json [hunk 8]
```diff
@@ -0,0 +1,13 @@
...
+        { "key": "Access-Control-Allow-Headers", "value": "X-CSRF-Token, X-Requested-With, Accept, Accept-Version, Content-Length, Content-MD5, Content-Type, Date, X-Api-Version" }
```
Comment (tusbar) [inline vercel.json]: Are these all necessary?

**Review by tusbar — CHANGES_REQUESTED** [review]

**Review by tusbar — APPROVED** [review]

**Pull request merged**